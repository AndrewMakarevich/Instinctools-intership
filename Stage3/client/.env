REACT_APP_BASE_BACK_URL=http://localhost:5000/api