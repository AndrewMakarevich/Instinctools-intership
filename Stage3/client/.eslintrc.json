{
  "env": {
    "browser": true,
    "es2021": true,
    "jest/globals": true
  },
  "extends": ["plugin:react/recommended", "airbnb", "prettier"],
  "parserOptions": {
    "ecmaFeatures": {
      "jsx": true
    },
    "ecmaVersion": "latest",
    "sourceType": "module"
  },
  "plugins": ["react", "jest", "prettier"],
  "rules": {
    "react/function-component-definition": 0,
    "no-underscore-dangle": 0,
    "jsx-quotes": 0,
    "react/forbid-prop-types": 0,
    "react/require-default-props": 0,
    "no-alert": 0,
    "jsx-a11y/label-has-associated-control": 0,
    "jsx-a11y/control-has-associated-label": 0,
    "jsx-a11y/click-events-have-key-events": 0,
    "jsx-a11y/no-static-element-interactions": 0,
    "operator-linebreak": 0,
    "object-curly-newline": 0,
    "default-param-last": 0,
    "import/no-extraneous-dependencies": 0,
    "react/jsx-props-no-spreading": 0,
    "import/no-cycle": 1,
    "comma-dangle": 1,
    "react/jsx-filename-extension": [1, { "extensions": ["js", "jsx"] }]
  }
}
