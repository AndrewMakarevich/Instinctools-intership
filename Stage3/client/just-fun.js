/* eslint-disable consistent-return */
/* eslint-disable no-param-reassign */
/* eslint-disable no-shadow */
/* eslint-disable no-unused-vars */
/* eslint-disable no-plusplus */
function comparison(firstEntity, secEntity) {
  let comparisonResult = true;

  function deepComparison(firstObj, secondObj) {
    if (!comparisonResult) {
      return;
    }

    if (Array.isArray(firstObj) && Array.isArray(secondObj)) {
      firstObj.forEach((item, index) => {
        deepComparison(item, secondObj[index]);
      });
      return;
    }

    if (typeof firstObj === 'object' && typeof secondObj === 'object') {
      Object.keys(firstObj).forEach((objectParam) => {
        deepComparison(firstObj[objectParam], secondObj[objectParam]);
      });
      return;
    }

    if (String(firstObj) === String(secondObj)) {
      return;
    }

    comparisonResult = false;
  }

  deepComparison(firstEntity, secEntity);
  return comparisonResult;
}

function copy(objectToCopy) {
  let copiedObject;

  if (Array.isArray(objectToCopy)) {
    copiedObject = [];
  }

  if (typeof objectToCopy === 'object') {
    copiedObject = {};
  }

  function createObjectCopy(objectToCopy) {
    if (Array.isArray(objectToCopy)) {
      const copiedArr = [];
      objectToCopy.forEach((item, index) => {
        copiedArr.push(createObjectCopy(item));
      });
      return copiedArr;
    }

    if (typeof objectToCopy === 'object') {
      const copiedObject = {};
      Object.keys(objectToCopy).forEach((param) => {
        copiedObject[param] = createObjectCopy(objectToCopy[param]);
      });
      return copiedObject;
    }
    return objectToCopy;
    // if(typeof objectToCopy === "object"){
    //   copiedObject = {}
    // }
  }

  createObjectCopy(objectToCopy, copiedObject);
  return copiedObject;
}

// const firstObject = {
//   name: 'Andrew',
//   secName: 2 + 2 + 2,
//   firstFunc: (number) => number + 1,
//   testArr: [1, 2, 3, { a: 1, b: 1 }],
//   arr: [{ a: 5, b: 6 }, { v: [1, 2, 3] }],
// };

const testArr = [1, 2, { a: 1, b: 2 }];

const copiedObject = copy(testArr);

console.log(copiedObject);

// const secondObject = {
//   name: 'Andrew',
//   secName: 2 * 3,
//   firstFunc: (number) => number + 1,
//   arr: [{ a: 5, b: 6 }, { v: [1, 2, 4] }],
// };

// console.log(comparison(firstObject, secondObject));
