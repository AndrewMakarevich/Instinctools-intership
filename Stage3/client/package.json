{
  "name": "client",
  "version": "0.1.0",
  "private": true,
  "dependencies": {
    "axios": "^0.26.1",
    "history": "^5.3.0",
    "prop-types": "^15.8.1",
    "react": "^18.0.0",
    "react-dom": "^18.0.0",
    "react-redux": "^7.2.8",
    "react-router-dom": "^6.3.0",
    "react-scripts": "5.0.0",
    "redux": "^4.1.2",
    "redux-thunk": "^2.4.1",
    "uuid": "^8.3.2",
    "web-vitals": "^2.1.4"
  },
  "scripts": {
    "start": "react-scripts start",
    "build": "react-scripts build",
    "test": "react-scripts test --coverage",
    "eject": "react-scripts eject"
  },
  "browserslist": {
    "production": [
      ">0.2%",
      "not dead",
      "not op_mini all"
    ],
    "development": [
      "last 1 chrome version",
      "last 1 firefox version",
      "last 1 safari version"
    ]
  },
  "devDependencies": {
    "@testing-library/jest-dom": "^5.16.4",
    "@testing-library/react": "^13.1.1",
    "@testing-library/user-event": "^14.1.1",
    "@types/jest": "^27.4.1",
    "eslint": "^8.14.0",
    "eslint-config-airbnb": "^19.0.4",
    "eslint-config-prettier": "^8.5.0",
    "eslint-config-react-app": "^7.0.1",
    "eslint-plugin-import": "^2.26.0",
    "eslint-plugin-jest": "^26.1.5",
    "eslint-plugin-jsx-a11y": "^6.5.1",
    "eslint-plugin-prettier": "^4.0.0",
    "eslint-plugin-react": "^7.29.4",
    "eslint-plugin-react-hooks": "^4.5.0",
    "install": "0.13.0",
    "npm": "8.8.0",
    "prettier": "2.6.2"
  }
}
