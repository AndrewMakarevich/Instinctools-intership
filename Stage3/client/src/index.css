* {
  box-sizing: border-box;
}

ul,
li,
h1,
h2,
h3,
h4 {
  margin: 0;
  padding: 0;
}

body {
  height: 100vh;
  min-height: 100vh;
  margin: 0;
  font-family: -apple-system, BlinkMacSystemFont, 'Segoe UI', 'Roboto', 'Oxygen',
    'Ubuntu', 'Cantarell', 'Fira Sans', 'Droid Sans', 'Helvetica Neue',
    sans-serif;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
}

code {
  font-family: source-code-pro, Menlo, Monaco, Consolas, 'Courier New',
    monospace;
}

#root {
  --extra-dark: #463F3A;
  --medium-dark: #8A817C;
  --medium-light: #BCB8B1;
  --extra-light: #F4F3EE;

  height: 100%;
  background-color: var(--medium-light);
}