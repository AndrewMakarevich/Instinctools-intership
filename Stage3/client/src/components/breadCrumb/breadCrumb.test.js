import { render, screen } from '@testing-library/react';
import { createRoot } from 'react-dom/client';
import BreadCrumb from './breadCrumb';
import AppRouter from '../router/appRouter';
import { BrowserRouter } from 'react-router-dom';

test('Bread crumb renders correctly', () => {
  render(
    <BrowserRouter>
      <BreadCrumb />
    </BrowserRouter>
  );
  const breadCrumbComponent = screen.getByTestId('bread-crumb');
  expect(breadCrumbComponent).toBeInTheDocument();
  expect(breadCrumbComponent).toMatchSnapshot(
    'Bread crumb: component final version'
  );
});
