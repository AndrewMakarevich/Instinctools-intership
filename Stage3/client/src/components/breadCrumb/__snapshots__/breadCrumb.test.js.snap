// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Bread crumb renders correctly: Bread crumb: component final version 1`] = `
<article
  class="crumb-wrapper"
  data-testid="bread-crumb"
>
  <div
    class="crumb-link__item"
  >
    <a
      class="link crumb-link"
      href="/"
    />
  </div>
</article>
`;
